packages/core/database/src/utils/remove-orphan-morph-types.ts

```diff
@@ -0,0 +1,70 @@
+import type { Database } from '..';
+
+/**
+ * Removes morph relation data with invalid or non-existent morph types.
+ *
+ * This function iterates over the database metadata to identify morph relationships
+ * (relations with a `joinTable` containing the specified pivot column) and removes
+ * any entries in the relation's join table where the morph type is invalid:
+ * - Morph types without corresponding metadata.
+ * - Morph types pointing to tables that no longer exist in the database.
+ *
+ * Note: This function does not check for orphaned IDs, only orphaned morph types.
+ *
+ * @param db - The database object containing metadata and a Knex connection.
+ * @param pivot - The name of the column in the join table representing the morph type.
+ */
+export const removeOrphanMorphTypes = async (db: Database, pivot: string) => {
+  const allTables = await db.dialect.schemaInspector.getTables();
```

Comment:
Could this depend on the memory schema directly instead of calling the getTables again ?

Reply:
TBH, it's not even fully necessary to do this; if something is not in the metadata, that's enough to know it should be deleted. I could remove this part.

Reply:
👍